src/commands/publish/mod.rs

```diff
@@ -258,7 +258,7 @@ fn validate_target_required_fields_present(target: &Target) -> Result<(), failur
         None => {}
     }
 
-    let destination = if target.route.is_some() {
+    let destination = if let Some(route) = &target.route {
```

Comment:
i could fix this to check `if target.routes()?.is_empty()`, however i'm also inclined to shift some of this logic into something like a `Manifest::get_deploy_config` function instead, so we can keep all the logic around what is needed to publish in one place; right now it is spread across this module and `settings::target`